plugins/filter/resource_list_compare.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: resource_list_compare
+    author: Rohit Thakur (@rohitthakur2590)
+    version_added: "2.4.0"
+    short_description: Generate the final resource list combining/comparing base and provided resources
+    description:
+        - Generate the final list of resources after comparing with base list and provide list of resources/bangs.
+    options:
+      base:
+        description: Specify the base list of supported network resources.
```

Comment:
Since the functionality of the plugin is not specific to the network and can be applied to facts in general how about renaming it to something genric like `facts_subset_compare` (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/setup_module.html#parameter-gather_subset)

Reply:
IMO param_list_compare sounds more generic.